parsers/basic.c

```diff
@@ -114,7 +114,7 @@ static int extract_dim (char const *pos, vString * name, BasicKind kind)
 		/* now we are at the name */
 	}
 	/* capture "dim as foo ptr bar" */
-	if (strncasecmp (pos, "ptr", 3) == 0 && isspace(*(pos+4)))
+	if (strncasecmp (pos, "ptr", 3) == 0 && isspace(*(pos+3)))
```

Comment:
I think this change should be merged to the commit " basic: add more advanced extraction of dim-like variables".